engine/src/main/java/org/terasology/rendering/opengl/GLSLMaterial.java

```diff
@@ -72,11 +73,13 @@
     private ShaderParameters shaderParameters;
 
     private DisposalAction disposalAction;
+    private MaterialData materialData;
 
     public GLSLMaterial(ResourceUrn urn, AssetType<?, MaterialData> assetType, MaterialData data) {
         super(urn, assetType);
         disposalAction = new DisposalAction(urn);
         getDisposalHook().setDisposeAction(disposalAction);
+        this.materialData = data;
         shaderManager = CoreRegistry.get(ShaderManager.class);
```

Comment:
In that case I think the best way would be the constructor. The instancing class has already a Context field it could pass to it.